### PR TITLE
improved form validation Fixes #92

### LINKED ISSUE #92: There is no proper handling of validation
Need to improve in the validation while registering to the website

### PR DESCRIPTION
**Pull request by venurockss**
I had added the form validation completely so that there is improper password while registering to the website ,so this implementation help increasing in the security.

### CODE REVIEW

File: views/users/register.ejs [hunk 1]
```diff
@@ -17,16 +17,29 @@
                         </div>
                         <div class="mb-3">
                             <label class="form-label" for="email">Email</label>
-                            <input class="form-control" type="email" id="email" name="email" required>
+                            <!-- Adding pattern for Gmail validation -->
+                            <input class="form-control" type="email" id="email" name="email" required
+                                pattern="^[a-zA-Z0-9._%+-]+@gmail\.com$" 
+                                title="Please enter a valid Gmail address (e.g., example@gmail.com).">
```
Comment (Vignesh025) [inline views/users/register.ejs:23]: The email field should accept all types of valid mail ids and not just gmail

File: views/users/register.ejs [hunk 2]
```diff
@@ -17,16 +17,29 @@
...
                             <div class="valid-feedback">
                                 Looks good!
                             </div>
+                            <div class="invalid-feedback">
+                                Please enter a valid Gmail address (e.g., example@gmail.com).
```
Comment (Vignesh025) [inline views/users/register.ejs:28]: Any mail id not just gmail

File: controllers/campgrounds.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 const Campground=require('../models/campground');
 const mbxGeocoding = require("@mapbox/mapbox-sdk/services/geocoding");
-const mapBoxToken=process.env.MAPBOX_TOKEN;
-const geocoder = mbxGeocoding({accessToken:mapBoxToken});
+// const mapBoxToken=process.env.MAPBOX_TOKEN;
+// const geocoder = mbxGeocoding({accessToken:mapBoxToken});
```
Comment (Vignesh025) [inline controllers/campgrounds.js:4]: Uncomment these two lines

**Review by Vignesh025 — COMMENTED** [review]
Make all the necessary changes.

**Comment by venurockss** [comment]
Okay

**Comment by Vignesh025** [comment]
Please remove 'gmail' and replace it with 'email' in the 28th line of register.ejs.

**Comment by Vignesh025** [comment]
Hey @venurockss , sync your fork and make the PR to the test branch and not the main branch

**Comment by venurockss** [comment]
@Vignesh025 can you please help me for steps to sunc my fork and make a pr to the test branch

**Comment by Vignesh025** [comment]
Click on 'Sync Fork' in your repository, then make a pull request with the base branch set to the 'test' branch of the project and the compare branch as your own branch.

**Comment by venurockss** [comment]
@Vignesh025 thank you , just created PR

**Pull request closed without merging**